pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -1716,8 +1716,54 @@ def clear(self):
         self.curve.clear()
         self.scatter.clear()
 
-    def appendData(self, *args, **kwargs):
-        pass
+
+    def appendData(self, new_x, new_y):
+        """
+        Append new data to the existing dataset.
+
+        .. warning::
+        This method may be significantly slower than calling `setData`, as it 
+        appends to the current data rather than replacing it. For high-performance
+        applications, consider using `setData` when possible.
+
+        .. warning::
+            This method will be undergoing further changes in future versions. 
+            Currently, the accepted arguments are only a subset of those for `setData`.
+            Ensure compatibility by using only the supported arguments.
+    
+        
+        Parameters
+        ----------
+        new_x : array-like
+            The new x-values to append.
+        new_y : array-like
+            The new y-values to append.
+        
+        """
+        if self._dataset is None:
+            # If there is no existing dataset, set the new data as the dataset.
+            self.setData(new_x, new_y)
+            return
+        
+        # Convert new data to arrays
+        new_x = np.array(new_x)
+        new_y = np.array(new_y)
+
+        # Concatenate new data to existing dataset
+        xData = np.concatenate((self._dataset.x, new_x))
+        yData = np.concatenate((self._dataset.y, new_y))
+
+            # Update the internal dataset with the concatenated arrays
+        self._dataset.x = xData
```

Comment:
Wouldn't it be more robust to just call `setData()` with the new arrays?
Currently it isn't entirely clear to me if the remaining code in `setData()`, such as emitting "sigPlotChanged()" are safe to omit in this case.

Reply:
Calling setData() would indeed ensure consistency by running all the additional code, like sigPlotChanged() and other updates that setData handles. However, in this case, as requested by @Khanhlinhdang  in issue #3154 the purpose of appendData() is to optimize for performance, especially for live data scenarios where frequent updates are required.

That said, appendData might not fully replicate all internal updates from setData for certain plot settings, so users should still rely on setData() when working with static datasets or when requiring a complete refresh.

Thus I have revised the docstring accordingly by adding the warnings as suggested by @j9ac9k

Reply:
How much performance improvement are you seeing by NOT going through `setData()`?

Reply:
Oh, maybe we are just misunderstanding each other?

I meant to say that you could call 'setData()' *here*, in line 1757 of your PR. That would avoid duplicating the cache clears etc. 

That then runs all the processing inside of pyqtgraph as usual. But the PR is not trying to avoid that. The PR is trying to help reduce processing *on the user side*. Do we agree on this? Or am I still getting it wrong?

Reply:
Yes, its clear

Reply:
Can I then once again suggest to replace L 1756 and the following lines with
`self.setData(xData, yData)` 
?
Would that not achieve the same thing with less duplication of code?